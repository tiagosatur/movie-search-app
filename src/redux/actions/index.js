export * from './movies';
export * from './details';