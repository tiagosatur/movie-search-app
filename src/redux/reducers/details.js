import { 
    MOVIE_DETAILS_PENDING,
    MOVIE_DETAILS_SUCCESS,
    MOVIE_DETAILS_ERROR,
} from '../../utils/actionTypes';

export const initialState = {
    pending: false,
    details: {},
    error: null
}

export default ( state = initialState, action ) => {
    const { type, details, error } = action;

    switch(type) {
        case MOVIE_DETAILS_PENDING:
            return {
                ...state,
                details: {},
                pending: true
            };
        case MOVIE_DETAILS_SUCCESS:
            return {
                ...state,
                pending: false,
                details: details,
            };
        case MOVIE_DETAILS_ERROR:
            return {
                ...state,
                details: {},
                pending: false,
                error: error
            };
        default: 
            return state;
    }
    
};