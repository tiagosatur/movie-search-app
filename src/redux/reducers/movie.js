import { 
    SEARCH_MOVIE_PENDING,
    SEARCH_MOVIE_SUCCESS,
    SEARCH_MOVIE_ERROR,
    SEARCH_MOVIE_CLEAR,
} from '../../utils/actionTypes';

export const initialState = {
    pending: false,
    movie: [],
    totalResults: 0,
    error: null
}

export default ( state = initialState, action ) => {
    const { type, movie, error } = action;

    switch(type) {
        case SEARCH_MOVIE_PENDING:
            return {
                ...state,
                movie: [],
                totalResults: 0,
                pending: true
            };
        case SEARCH_MOVIE_CLEAR :
            return {
                ...state,
                movie: [],
                totalResults: 0,
                pending: true
            };
        case SEARCH_MOVIE_SUCCESS:
            return {
                ...state,
                pending: false,
                movie: movie.Search,
                totalResults: movie.totalResults,
            };
        case SEARCH_MOVIE_ERROR:
            return {
                ...state,
                movie: [],
                pending: false,
                totalResults: 0,
                error: error
            };
        default: 
            return state;
    }
};