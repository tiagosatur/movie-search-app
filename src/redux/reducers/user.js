import { 
    LOGIN_USER_PENDING,
    LOGIN_USER_SUCCESS,
    LOGIN_USER_ERROR,
} from '../../utils/actionTypes';

export const initialState = {
    pending: false,
    isLogged: false,
    user: {},
    error: null
}

export const user = ( state = initialState, action ) => {
    const { type, user, error } = action;

    switch(type) {
        case LOGIN_USER_PENDING:
            return {
                ...state,
                user: {},
                pending: true
            };
        case LOGIN_USER_SUCCESS:
            return {
                ...state,
                pending: false,
                isLogged: true,
                user: user,
            };
        case LOGIN_USER_ERROR:
            return {
                ...state,
                pending: false,
                isLogged: false,
                user: {},
                error: error
            };
        default: 
            return state;
    }
};