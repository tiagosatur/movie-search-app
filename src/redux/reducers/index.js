import { combineReducers } from 'redux'
import {movie} from './movie'
import {details} from './details'
import {user} from './user'

export default combineReducers ({
    movie,
    details,
    user,
})