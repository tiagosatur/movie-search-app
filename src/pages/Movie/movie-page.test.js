import React from 'react';
import {  Provider } from 'react-redux'
import { initialState } from '../../redux/reducers/movie';
import { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import ConnectedMovie, { Movie, StyledMovieItem } from './Movie';
import searchMovieApiMock from '../../mocks/searchMovieApiMock';
import {  MemoryRouter, StaticRouter } from "react-router";
import { createStore } from 'redux';
import rootReducer from '../../redux/reducers'

describe('Movie Page tests', () => {
    let store, wrapper;

    beforeEach(() => {
        store = createStore(rootReducer, searchMovieApiMock);
        wrapper =  mount(
            <Provider store={store}>
                <MemoryRouter initialEntries={[ '/movies' ]}>
                    <ConnectedMovie />
                </MemoryRouter>
            </Provider>
        );
    });

    it('should render the Search Form', () => {
        expect(wrapper.find('.search-form').exists()).toBe(true);
    });

    it('should call Movie snapshot correctly', () => {
        const tree = renderer
            .create(wrapper)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
})

