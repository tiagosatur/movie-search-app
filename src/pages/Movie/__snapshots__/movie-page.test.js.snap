// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Page tests should call Movie snapshot correctly 1`] = `
<div
  className="movies"
>
  <div>
    <form
      className="search-form sc-kkGfuU lciZVM"
      onSubmit={[Function]}
    >
      <input
        className="input  sc-bZQynM PbcdM"
        name="searchValue"
        onChange={[Function]}
        placeholder="Search a movie..."
        value=""
      />
      <button
        className="button  sc-bdVaJa cbsQlW"
        title=""
        type="submit"
      >
        Search
      </button>
    </form>
  </div>
  <div
    className="sc-hMqMXs hIuDJf"
  >
    <div
      className="modal sc-VigVT jQagwL"
    >
      <div
        className="sc-jTzLTM jZBWoM"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-window-close fa-w-16 sc-gqjmRU fsxKTh"
          data-icon="window-close"
          data-prefix="fas"
          focusable="false"
          onClick={[Function]}
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M464 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h416c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-83.6 290.5c4.8 4.8 4.8 12.6 0 17.4l-40.5 40.5c-4.8 4.8-12.6 4.8-17.4 0L256 313.3l-66.5 67.1c-4.8 4.8-12.6 4.8-17.4 0l-40.5-40.5c-4.8-4.8-4.8-12.6 0-17.4l67.1-66.5-67.1-66.5c-4.8-4.8-4.8-12.6 0-17.4l40.5-40.5c4.8-4.8 12.6-4.8 17.4 0l66.5 67.1 66.5-67.1c4.8-4.8 12.6-4.8 17.4 0l40.5 40.5c4.8 4.8 4.8 12.6 0 17.4L313.3 256l67.1 66.5z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
